www/js/app.js

```diff
@@ -1017,6 +1021,14 @@ define(['jquery', 'zimArchiveLoader', 'util', 'uiUtil', 'cookies','abstractFiles
         // but we should not do this when opening the landing page (or else one of the Unit Tests fails, at least on Chrome 58)
         if (!params.isLandingPage) document.getElementById('articleContent').contentWindow.focus();
 
+        // Early return if article's title not equal current url's title
```

Comment:
I think this piece of code should be moved right before line `iframeArticleContent.src = `

Reply:
Hi mossroy, I have some trouble with the `selectedArchive.resolveRedirect(dirEntry, readArticle);` part in the `readArticle()` function. 

For now I just update the global variable `expectedArticleURLToBeDisplayed`(I exchanged the name of curTitle and expectedArticleURLToBeDisplayed as suggested from your previous comment) everytime before calling `readArticle()`.  In the case of viewing documents where there is a redirect,  `selectedArchive.resolveRedirect(dirEntry, readArticle);` will trigger the `readArticle()` without updating the `expectedArticleURLToBeDisplayed` and cause an early return. Is setting another global variable `isRedirectFromOtherEntry` gonna to help with fixing this unexpected early return caused by redirecting? Say `isRedirectFromOtherEntry` is set to true every time the `resolveRedirect()` is called and the `readArticle()` will skip early return if `isRedirectFromOtherEntry` is true and set it back to false.

Reply:
Could you please push to github the current state of your code? I do not see the variable names exchanged, for example.
I don't think we should introduce a new global variable for that. It would be better that a redirect would update the global variable to the new URL, before starting reading it in the backend

Reply:
Code updated. I tested my version locally and it seems fine now.